// roarr/dist/factories/createRoarrInititialGlobalState.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _semverCompare = _interopRequireDefault(require("semver-compare"));

var _package = require("../../package.json");

var _config = require("../config");

var _createWriter = _interopRequireDefault(require("./createWriter"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

// eslint-disable-next-line flowtype/no-weak-types
var createRoarrInititialGlobalState = function createRoarrInititialGlobalState(currentState) {
  var versions = (currentState.versions || []).concat();
  versions.sort(_semverCompare.default);
  var currentIsLatestVersion = !versions.length || (0, _semverCompare.default)(_package.version, versions[versions.length - 1]) === 1;

  if (!versions.includes(_package.version)) {
    versions.push(_package.version);
  }

  versions.sort(_semverCompare.default);

  var newState = _objectSpread({
    buffer: '',
    prepend: {},
    sequence: 0
  }, currentState, {
    versions
  });

  if (currentIsLatestVersion || !newState.write) {
    newState = _objectSpread({}, newState, (0, _createWriter.default)({
      bufferSize: _config.ROARR_BUFFER_SIZE,
      stream: _config.ROARR_STREAM
    }));
  }

  return newState;
};

var _default = createRoarrInititialGlobalState;
exports.default = _default;
