// slonik/dist/binders/bindPoolConnection.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _utilities = require("../utilities");

var _connectionMethods = require("../connectionMethods");

const bindPoolConnection = (parentLog, connection, clientConfiguration) => {
  return {
    any: (0, _utilities.mapTaggedTemplateLiteralInvocation)(_connectionMethods.any.bind(null, parentLog, connection, clientConfiguration)),
    anyFirst: (0, _utilities.mapTaggedTemplateLiteralInvocation)(_connectionMethods.anyFirst.bind(null, parentLog, connection, clientConfiguration)),
    many: (0, _utilities.mapTaggedTemplateLiteralInvocation)(_connectionMethods.many.bind(null, parentLog, connection, clientConfiguration)),
    manyFirst: (0, _utilities.mapTaggedTemplateLiteralInvocation)(_connectionMethods.manyFirst.bind(null, parentLog, connection, clientConfiguration)),
    maybeOne: (0, _utilities.mapTaggedTemplateLiteralInvocation)(_connectionMethods.maybeOne.bind(null, parentLog, connection, clientConfiguration)),
    maybeOneFirst: (0, _utilities.mapTaggedTemplateLiteralInvocation)(_connectionMethods.maybeOneFirst.bind(null, parentLog, connection, clientConfiguration)),
    one: (0, _utilities.mapTaggedTemplateLiteralInvocation)(_connectionMethods.one.bind(null, parentLog, connection, clientConfiguration)),
    oneFirst: (0, _utilities.mapTaggedTemplateLiteralInvocation)(_connectionMethods.oneFirst.bind(null, parentLog, connection, clientConfiguration)),
    query: (0, _utilities.mapTaggedTemplateLiteralInvocation)(_connectionMethods.query.bind(null, parentLog, connection, clientConfiguration)),
    stream: async (streamQuery, streamHandler) => {
      if (typeof streamQuery === 'string') {
        throw new TypeError('Query must be constructed using `sql` tagged template literal.');
      }

      return (0, _connectionMethods.stream)(parentLog, connection, clientConfiguration, streamQuery.sql, streamQuery.values, streamHandler);
    },
    transaction: async handler => {
      return (0, _connectionMethods.transaction)(parentLog, connection, clientConfiguration, handler);
    }
  };
};

var _default = bindPoolConnection;
exports.default = _default;
