// slonik/dist/errors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CheckIntegrityConstraintViolationError = exports.UniqueIntegrityConstraintViolationError = exports.ForeignKeyIntegrityConstraintViolationError = exports.NotNullIntegrityConstraintViolationError = exports.IntegrityConstraintViolationError = exports.DataIntegrityError = exports.NotFoundError = exports.BackendTerminatedError = exports.QueryCancelledError = exports.ConnectionError = exports.UnexpectedStateError = exports.SlonikError = void 0;

var _es6Error = _interopRequireDefault(require("es6-error"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable fp/no-class, fp/no-this */
class SlonikError extends _es6Error.default {}

exports.SlonikError = SlonikError;

class UnexpectedStateError extends SlonikError {}

exports.UnexpectedStateError = UnexpectedStateError;

class ConnectionError extends SlonikError {}

exports.ConnectionError = ConnectionError;

class QueryCancelledError extends SlonikError {}

exports.QueryCancelledError = QueryCancelledError;

class BackendTerminatedError extends SlonikError {}

exports.BackendTerminatedError = BackendTerminatedError;

class NotFoundError extends SlonikError {
  constructor() {
    super('Resource not found.');
  }

}

exports.NotFoundError = NotFoundError;

class DataIntegrityError extends SlonikError {
  constructor() {
    super('Query returns an unexpected result.');
  }

}

exports.DataIntegrityError = DataIntegrityError;

class IntegrityConstraintViolationError extends SlonikError {
  constructor(constraint) {
    super();
    this.constraint = constraint;
  }

} // @todo When does restrict_violation and exclusion_violation happen?
// @see https://www.postgresql.org/docs/9.4/static/errcodes-appendix.html


exports.IntegrityConstraintViolationError = IntegrityConstraintViolationError;

class NotNullIntegrityConstraintViolationError extends IntegrityConstraintViolationError {
  constructor(constraint) {
    super(constraint);
    this.message = 'Query violates a not NULL integrity constraint.';
  }

}

exports.NotNullIntegrityConstraintViolationError = NotNullIntegrityConstraintViolationError;

class ForeignKeyIntegrityConstraintViolationError extends IntegrityConstraintViolationError {
  constructor(constraint) {
    super(constraint);
    this.message = 'Query violates a foreign key integrity constraint.';
  }

}

exports.ForeignKeyIntegrityConstraintViolationError = ForeignKeyIntegrityConstraintViolationError;

class UniqueIntegrityConstraintViolationError extends IntegrityConstraintViolationError {
  constructor(constraint) {
    super(constraint);
    this.message = 'Query violates a unique integrity constraint.';
  }

}

exports.UniqueIntegrityConstraintViolationError = UniqueIntegrityConstraintViolationError;

class CheckIntegrityConstraintViolationError extends IntegrityConstraintViolationError {
  constructor(constraint) {
    super(constraint);
    this.message = 'Query violates a check integrity constraint.';
  }

}

exports.CheckIntegrityConstraintViolationError = CheckIntegrityConstraintViolationError;
