// slonik/dist/symbols.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UnnestTokenSymbol = exports.TupleListTokenSymbol = exports.TupleTokenSymbol = exports.ValueListTokenSymbol = exports.IdentifierListTokenSymbol = exports.IdentifierTokenSymbol = exports.RawSqlTokenSymbol = exports.SqlTokenSymbol = exports.ArrayTokenSymbol = void 0;
const ArrayTokenSymbol = Symbol('SLONIK_ARRAY');
exports.ArrayTokenSymbol = ArrayTokenSymbol;
const SqlTokenSymbol = Symbol('SLONIK_SQL');
exports.SqlTokenSymbol = SqlTokenSymbol;
const RawSqlTokenSymbol = Symbol('SLONIK_RAW_SQL');
exports.RawSqlTokenSymbol = RawSqlTokenSymbol;
const IdentifierTokenSymbol = Symbol('SLONIK_IDENTIFIER');
exports.IdentifierTokenSymbol = IdentifierTokenSymbol;
const IdentifierListTokenSymbol = Symbol('SLONIK_IDENTIFIER_LIST');
exports.IdentifierListTokenSymbol = IdentifierListTokenSymbol;
const ValueListTokenSymbol = Symbol('SLONIK_VALUE_LIST');
exports.ValueListTokenSymbol = ValueListTokenSymbol;
const TupleTokenSymbol = Symbol('SLONIK_TUPLE');
exports.TupleTokenSymbol = TupleTokenSymbol;
const TupleListTokenSymbol = Symbol('SLONIK_TUPLE_LIST');
exports.TupleListTokenSymbol = TupleListTokenSymbol;
const UnnestTokenSymbol = Symbol('SLONIK_UNNEST');
exports.UnnestTokenSymbol = UnnestTokenSymbol;
